File: programmers/js/level00/120844.js
// 배열 회전시키기
function solution(numbers, direction) {
  switch (direction) {
    case 'right':
      return [numbers.at(-1), ...numbers.slice(0, -1)];
    case 'left':
      return [...numbers.slice(1), numbers[0]];
  }
}

File: programmers/java/level00/120844.java
// 배열 회전시키기

import java.util.ArrayList;

class Solution {
    public int[] solution(int[] numbers, String direction) {
        ArrayList<Integer> result = new ArrayList<>();
        
        if (direction.equals("right")) {
            result.add(numbers[numbers.length - 1]);
            
            for (int i = 0; i < numbers.length - 1; i++) {
                result.add(numbers[i]);
            }
        } else if (direction.equals("left")) {
            for (int i = 1; i < numbers.length; i++) {
                result.add(numbers[i]);
            }
            
            result.add(numbers[0]);
        }
        
        return result.stream().mapToInt(v -> v).toArray();
    }
}